src/zarr/storage/_object.py

```diff
@@ -73,20 +71,21 @@ def __init__(self, store: _ObjectStore, *, read_only: bool = False) -> None:
             ),
         ):
             raise TypeError(f"expected ObjectStore class, got {store!r}")
-        self.store = store
         super().__init__(read_only=read_only)
+        self.store = store
 
     def __str__(self) -> str:
         return f"object://{self.store}"
 
     def __repr__(self) -> str:
         return f"ObjectStore({self})"
 
-    def __getstate__(self) -> None:
-        raise NotImplementedError("Pickling has not been implement for ObjectStore")
+    def __getstate__(self) -> dict:
+        self.__dict__.update({"store": pickle.dumps(self.store)})
```

Comment:
That looks great. Does `self.__dict__` already contain the value of `read_only`?

Reply:
I think so, since `assert new_store.read_only == store.read_only` passes